src/cryptonote_protocol/cryptonote_protocol_handler.inl

```diff
@@ -308,7 +308,8 @@ namespace cryptonote
       << " [Your node is " << abs_diff << " blocks (" << ((abs_diff - diff_v2) / (24 * 60 * 60 / DIFFICULTY_TARGET_V1)) + (diff_v2 / (24 * 60 * 60 / DIFFICULTY_TARGET_V2)) << " days) "
       << (0 <= diff ? std::string("behind") : std::string("ahead"))
       << "] " << ENDL << "SYNCHRONIZATION started");
-      m_core.safesyncmode(false);
+      if (hshd.current_height - m_core.get_current_blockchain_height() >= 5) // don't switch to unsafe mode just for a few blocks
```

Comment:
Can't this potentially underflow? Why not use the var `diff` above?

Reply:
Indeed, fixing.